### ISSUE TITLE
NeuVector k8s 1.27 chart support

### ISSUE THREAD

**Issue by gunamata** [rancher/rancher#41803]
This issue is to implement the chart changes & perform the testing required to upgrade to k8s 1.27. Please perform below steps. 

- [ ] Check whether the chart has any references to the below deprecated features of k8s 1.27. You can find information about deprecated APIs here on [Kubernetes Deprecation guide](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-27).

**storage.k8s.io/v1beta1** API version of **CSIStorageCapacity**
 

- [ ] Update kube-version annotation in Chart.yaml. catalog.cattle.io/kube-version: '>= 1.23.0-0 < 1.28.0-0'

Note: If this is an upstream chart and if the upstream chart doesn't support k8s 1.27 then do not update the kube-version annotation or update the k8s version accordingly.

- [ ] Perform below tests

- Fresh install/uninstall/upgrade/downgrade of the charts on K8s 1.27 provisioned on Rancher. Test the chart functionality.
(Please refer to the https://github.com/rancher/rancher/issues/41395 to know what testing was done for k8s 1.26 upgrade for your chart)
- Provision rancher on k8s 1.26 or lesser
- Install chart on this cluster
- Upgrade this cluster in rancher to k8s 1.27
- Make sure the existing chart and the app is functional after cluster upgrade

**Comment by gunamata** [comment]
@selvamt94 , Are you blocked on something to start this task?

**Comment by selvamt94** [comment]
@gunamata We are not sure which Rancher Manager version need to be used. RM version 2.7.7(rc) is not supporting k8s version 1.28. I checked with RM QA team, I was told that this issue is for Q4.

**Comment by gunamata** [comment]
@selvamt94 , We are targeting k8s 1.27 only for the Rancher 2.8 release and deferred k8s 1.28 upgrade to Q1 2024. So, you just need test with k8s 1.27

**Comment by selvamt94** [comment]
@gunamata Sorry I mentioned wrong about RM 2.7.7(rc). RM version 2.7.7(rc) is not supporting k8s version 1.27. So do we need to verify on RM 2.8 dev branch?

**Comment by gunamata** [comment]
@selvamt94 , you need to test this with the RM 2.8 dev branch.

**Comment by selvamt94** [comment]
Verified NeuVector feature chart for k8s 1.27 version support. 

Deployed Rancher Manager v2.8-head
Created RKE2 cluster with k8s version 1.26 
Deployed Neuvector version 102.0.4+up2.6.2
Tested NeuVector deployment
Upgraded RKE2 cluster to k8s 1.27 version
Tested NeuVector deployment - worked fine.
